### PR TITLE
[quant][graph] Add _choose_qparams function for graph mode

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35366 [quant][graph] Add pass to insert quant dequant for dynamic quantization
* #35325 [quant][graph] Update dynamic quant tests to use new qconfig
* #35265 [quant][graph] Add a new observer type for dynamic quantization
* **#35235 [quant][graph] Add _choose_qparams function for graph mode**

Summary: For dynamic quantization in graph mode, we need an operator that returns the qparams of the tensor
similar to the linear_dynamic quantized op

Test Plan:
python test/test_quantized_tensor.py TestQuantizedTensor.test_choose_qparams

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20608793](https://our.internmc.facebook.com/intern/diff/D20608793)

### CODE REVIEW

File: torch/csrc/autograd/utils/wrap_outputs.h [hunk 1]
```diff
@@ -185,4 +185,13 @@ inline PyObject* wrap(at::IntArrayRef list) {
   }
   return r.release();
 }
+
+inline PyObject* wrap(std::tuple<float, int64_t> tensors) {
+  auto r = THPObjectPtr{PyTuple_New(5)};
```
Comment (albanD) [inline torch/csrc/autograd/utils/wrap_outputs.h]: Shouldn't this be `2`?

Comment (supriyar) [reply torch/csrc/autograd/utils/wrap_outputs.h]: Yeah, typo. Fixed it.

File: aten/src/ATen/native/quantized/QTensor.cpp [hunk 2]
```diff
@@ -218,5 +219,22 @@ bool quantized_equal(const Tensor& self, const Tensor& other) {
   return 0 == memcmp(self_data, other_data, self.numel() * self.element_size());
 }
 
+/* Calculate the quantization params for the activation tensor */
+std::tuple<double, int64_t> _choose_qparams(const Tensor& self, bool reduce_range) {
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/QTensor.cpp]: we'll need to have another function for per channel quantization as well until we expose Quantizer to python, maybe rename this to `_choose_qparams_per_tensor` now?

Comment (supriyar) [reply aten/src/ATen/native/quantized/QTensor.cpp]: Do we need per-channel quantization for activations though?

File: aten/src/ATen/native/quantized/QTensor.cpp [hunk 3]
```diff
@@ -218,5 +219,22 @@ bool quantized_equal(const Tensor& self, const Tensor& other) {
...
+  auto input_contig = self.contiguous();
+  float x_min = input_contig.min().item<float>();
+  float x_max = input_contig.max().item<float>();
+
+  auto q_params = quant_utils::ChooseQuantizationParams(
+        /*min=*/x_min,
+        /*max=*/x_max,
+        /*qmin=*/0,
+        /*qmax=*/255,
+        reduce_range=reduce_range);
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/QTensor.cpp]: really? can you pass keyword arguments in C++?

File: aten/src/ATen/native/quantized/QTensor.cpp (hunk 2 above)
Comment (jerryzh168) [reply aten/src/ATen/native/quantized/QTensor.cpp]: probably not, but this is still a `_choose_qparams` only for per tensor quantization, I feel it's better to be more explicit.

File: aten/src/ATen/native/quantized/QTensor.cpp (hunk 3 above)
Comment (supriyar) [reply aten/src/ATen/native/quantized/QTensor.cpp]: Ah no. It was supposed to be a comment. I'll update the PR

**Review by jerryzh168 — APPROVED** [review]
Looks good

**Pull request closed without merging**